### PR TITLE
feat : Added wishlist and cart functionality

### PR DESCRIPTION
**Pull request by BalajiHarinath**
I have added the wishlist and cart functionality. Please review.

Files to be reviewed:-
1. Context/CartContext.js
2. Context/WishlistContext.js
3. Context/authenticationContext.js
4. Pages/Cart/Cart.jsx
5. Pages/Cart/cartComputation.js
6. Pages/Cart/CartDetails.jsx
7. Pages/Cart/CartItemCard.jsx
8. Pages/Wishlist/Wishlist.jsx
9. Pages/Wishlist/WishlistItemCard.jsx
10. Pages/Reducers/authReducer.js

If you want to login without test credentials, please use the following credentials:
email: [adarshbalika@gmail.com](mailto:adarshbalika@gmail.com)
password: adarsh123

else you can sign up using your own credentials.

### CODE REVIEW

File: src/Components/Header/Nav.jsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export const Nav = () => {
                     <Link className="flex flex-column" to="/wishlist">
                         <span className="container-badge">
                             <span className="material-icons text-2xl nav-links nav-wishlist">favorite</span>
-                            <span className="status-badge status-badge-number">2</span>
+                            <span className={`${wishlist.length>0 ? "status-badge status-badge-number" : "display-none"}`}>{wishlist.length}</span>
```
Comment (ApurvaSawant11) [inline src/Components/Header/Nav.jsx]: You can take advantage of React here. You can achieve this without class toggling

`{wishlist.length>0 && <span className="status-badge status-badge-number">{wishlist.length}</span>}`

File: src/Pages/Authentication/Login.jsx [hunk 2]
```diff
@@ -32,10 +32,12 @@ export const Login = () => {
         }
 
     }
+    // console.log(error)
 
     return(
         <main className="main-login flex flex-justify-center flex-align-center mt-4 mb-4">
         <div className="form-login">
+            {/* {console.log(error)} */}
```
Comment (ApurvaSawant11) [inline src/Pages/Authentication/Login.jsx]: Remove comments if not needed

File: src/Pages/Wishlist/Wishlist.css [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    transition: all 0.1s ease;
+}
+
+.wishlist-card:hover {
+    transform: scale(1.01);
+}
+
+.container-img-wishlist {
+    /* width: 16rem;
+    height: 16.875rem;  */
```
Comment (ApurvaSawant11) [inline src/Pages/Wishlist/Wishlist.css]: Remove comments

**Review by ApurvaSawant11 — APPROVED** [review]
Looks good to me. Just gave one suggestion.

**Review by mikesch803 — APPROVED** [review]
just do format document else your code is good

File: src/Components/Header/Nav.jsx (hunk 1 above)
Comment (BalajiHarinath) [reply src/Components/Header/Nav.jsx]: yes got it, will change this. Thank you for suggesting this.

**Comment by BalajiHarinath** [comment]
I have made the suggested changes. Thanks for the review. Merging the PR.

**Pull request merged**